src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfig.java

```diff
@@ -237,19 +251,29 @@ private void validatePKChunk(FailureCollector collector) {
 
     if (!containsMacro(SalesforceSourceConstants.PROPERTY_QUERY) && !Strings.isNullOrEmpty(query)) {
       if (SalesforceQueryParser.isRestrictedPKQuery(query)) {
-          collector.addFailure(
-            String.format("SOQL Query contains restricted clauses when PK Chunk is Enabled. Unsupported query: '%s'.",
-                          query),
-            "Set Enable PK Chunk to false, because 'WHERE' is the only supported conditions clause.")
-            .withConfigProperty(SalesforceSourceConstants.PROPERTY_QUERY);
-        }
+        collector.addFailure(
+          String.format("SOQL Query contains restricted clauses when PK Chunk is Enabled. Unsupported query: '%s'.",
+                        query),
+          "Set Enable PK Chunk to false, because 'WHERE' is the only supported conditions clause.")
+          .withConfigProperty(SalesforceSourceConstants.PROPERTY_QUERY);
+      }
 
-      String sObject = SalesforceQueryParser.getObjectDescriptorFromQuery(query).getName();
-      checkForPKSupportedObject(sObject, collector);
+      // If a parent object is defined then use that to check for PK support, otherwise use the object from the query
+      if (!containsMacro(SalesforceSourceConstants.PROPERTY_PARENT_NAME) && !Strings.isNullOrEmpty(getParent())) {
+        checkForPKSupportedObject(getParent(), collector);
+      } else {
+        String sObject = SalesforceQueryParser.getObjectDescriptorFromQuery(query).getName();
+        checkForPKSupportedObject(sObject, collector);
```

Comment:
if parent name is a macro, then we're checking the object from the query instead of the parent object. Is that expected?

Reply:
if the parent field is a marco then it wont get to this point, the first check in this function exits if any of the properties are macros.

We technically dont need any of the marco checks after the first one but I left it in there for readability

Reply:
I see, thanks